hedera-node/hedera-mono-service/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -196,14 +199,27 @@ private void publishSyntheticCreation(
             final Key key,
             final String accountMemo,
             final String recordMemo,
-            final String description) {
+            final String description,
+            final long balance) {
         final var tracker = sideEffectsFactory.get();
         tracker.trackAutoCreation(num.toGrpcAccountId(), ByteString.EMPTY);
         final var synthBody =
                 synthCreation(
-                        autoRenewPeriod, key, accountMemo, receiverSigRequired, declineReward);
+                        autoRenewPeriod,
+                        key,
+                        accountMemo,
+                        receiverSigRequired,
+                        declineReward,
+                        balance);
         final var synthRecord =
                 creator.createSuccessfulSyntheticRecord(NO_CUSTOM_FEES, tracker, recordMemo);
+        // show the balance of treasury account for mirror node reconciliation.
+        if (num.longValue() == 2L) {
```

Comment:
Could use `if (balance != 0)` in case we end up supporting genesis networks where multiple accounts have a non-zero initial balance.

Reply:
Makes sense. Fixed. Thanks !